client/src/components/games/GamesPerVenue.tsx

```diff
@@ -1,24 +1,28 @@
 import { useContext, useEffect, useState } from "react";
 import { NavLink, useParams } from "react-router-dom";
+import { Game } from '../../@types/game';
+import { Venue } from '../../@types/venue';
 import { GamesContext, VenuesContext } from "../../App";
-import NavBar from "../common/NavBar";
 import { dateDisplay, durationDisplay, timeDisplay } from '../../utilities/date-time-display';
+import NavBar from "../common/NavBar";
 
 function GamesPerVenue() {
   const { games } = useContext(GamesContext)
-  const venues = useContext(VenuesContext)
+  const { venues } = useContext(VenuesContext)
   const { venueId } = useParams()
-  const [filteredGames, setFilteredGames] = useState([])
-  const [venue, setVenue] = useState({})
+  const [filteredGames, setFilteredGames] = useState<Game[] | null>([])
+  const [venue, setVenue] = useState<Venue | null>(null)
 
   useEffect(() => {
-    const filtered = games.filter(game => game.venue && game.venue._id === venueId).sort((a, b) => new Date(a.date) - new Date(b.date));
-    setFilteredGames(filtered)
-
-    const venue = venues.find(venue => venue._id === venueId)
-    setVenue(venue)
+    const filtered = games.filter(game => game.venue && game.venue.id === venueId).sort((a, b) => new Date(a.date).getTime() - new Date(b.date).getTime());
+    setFilteredGames(filtered);
   }, [games])
 
+  useEffect(() => {

```

Comment:
This handles changes in the currently viewed venue. Would it also handle the case where the current venue gets deleted?